datafusion/physical-expr/src/aggregate/approx_percentile_cont.rs

```diff
@@ -284,7 +284,9 @@ impl ApproxPercentileAccumulator {
     }
 
     pub(crate) fn merge_digests(&mut self, digests: &[TDigest]) {
-        self.digest = TDigest::merge_digests(digests);
+        let mut input_digests = digests.to_vec();
```

Comment:
As one Accumulator call  merge() should not lose it inner status, this is not a good API desgion.

Reply:
The change makes sense to me, but it is hard to review without a test that demonstrates the incorrect behavior. Would it be possible to add a unit test as part of this PR?

Reply:
100% agree we should add a test for this fix (otherwise we may break the behavior again in a subsequent refactoring, for example)